shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.mm

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.h"
+
+#import <OpenGL/gl.h>
+#import "flutter/fml/logging.h"
+#import "flutter/fml/platform/darwin/cf_utils.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/MacOSSwitchableGLContext.h"
+#import "third_party/skia/include/core/SkCanvas.h"
+#import "third_party/skia/include/core/SkSurface.h"
+#import "third_party/skia/include/gpu/gl/GrGLAssembleInterface.h"
+#import "third_party/skia/include/utils/mac/SkCGUtils.h"
+
+#include <unistd.h>
+
+namespace flutter {
+
+FlutterMacOSGLCompositor::FlutterMacOSGLCompositor(FlutterViewController* view_controller)
+    : view_controller_(view_controller) {
+  auto interface = GrGLMakeNativeInterface();
+  GrContextOptions options;
+  context_ = GrDirectContext::MakeGL(GrGLMakeNativeInterface(), options);
+
+  openGLContext_ =
+      [[NSOpenGLContext alloc] initWithFormat:view_controller_.flutterView.openGLContext.pixelFormat
+                                 shareContext:view_controller_.flutterView.openGLContext];
+}
+
+FlutterMacOSGLCompositor::~FlutterMacOSGLCompositor() = default;
+
+bool FlutterMacOSGLCompositor::CreateBackingStore(const FlutterBackingStoreConfig* config,
+                                                  FlutterBackingStore* backing_store_out) {
+  return CreateBackingStoreUsingSurfaceManager(config, backing_store_out);
+}
+
+bool FlutterMacOSGLCompositor::CollectBackingStore(const FlutterBackingStore* backing_store) {
+  // We have already set the destruction callback for the various backing
+  // stores. Our user_data is just the canvas from that backing store and does
+  // not need to be explicitly collected. Embedders might have some other state
+  // they want to collect though.
```

Comment:
` Embedders might have some other state they want to collect though.` is confusing. Can you elaborate?

Reply:
Removed this comment